core/network/impl/stream_engine.hpp

```diff
@@ -317,6 +320,36 @@ namespace kagome::network {
       broadcast(protocol, msg, any);
     }
 
+    template <typename T>
+    void gossipLuckyPeers(const std::shared_ptr<ProtocolBase> &protocol,
+                          const std::shared_ptr<T> &msg) {
+      if (lucky_peers_num_ > 0) {
+        std::deque<PeerId> peers;
+
+        forEachPeer([&](const auto &peer_id, auto &proto_map) {
```

Comment:
Maybe instead of obtaining peers using lambda and then broadcasting to them put this logic into new method `forLuckyPeer`?

Reply:
Or just generate a list of indexes which we intend to gossip to and check in forEachPeer whether we want to gossip to this peer.